pkg/resources/nifi/secretconfig.go

```diff
@@ -352,14 +357,18 @@ func (r *Reconciler) getLogbackConfigString(nConfig *v1.NodeConfig, id int32, lo
 	if r.NifiCluster.Spec.ReadOnlyConfig.LogbackConfig.ReplaceConfigMap != nil {
 		conf, err := r.getConfigMap(context.TODO(), *r.NifiCluster.Spec.ReadOnlyConfig.LogbackConfig.ReplaceConfigMap)
 		if err == nil {
-			return conf
+			return &conf
 		}
 		log.Error("error occurred during getting readonly configmap",
 			zap.String("clusterName", r.NifiCluster.Name),
 			zap.Int32("nodeId", id),
 			zap.Error(err))
 	}
 
+	if !configcommon.MustOverrideLogback(r.NifiCluster) {
```

Comment:
Since this always returns false, we'll never use the logback template in nifikop with this change. Should we just remove it?

Reply:
I don't, maybe in the futur we would need it? So is it better to keep it? I don't know.

Reply:
I feel like it's okay to allow overrides or just default to the logback that ships with NiFi. In that case, we shouldn't maintain a template within nifikop except as an example for how to override the defaults. Would you agree?

My opinion is to remove it and if we need to replace it, we can track down this PR to know what needs to be added. Do you agree?

Reply:
If I do it, should I do it also for `bootstrap_notifications_services.xml`?

And should I add `bootstrap-azure.conf`, `bootstrap-gcp.conf`, `bootstrap-hashicorp-vault.conf` and `bootstrap-aws.conf`?

Reply:
Perhaps we wait until there's an actual need to override those and implement that in a separate PR?

Reply:
Yes it will be done in different PRs. But as we are refactoring all config files, I thought that it could be interesting to address all of them.

So if we remove the template of the logback, I don't think there is a reason to keep the one for  `bootstrap_notifications_services.xml`.

And `bootstrap-azure.conf`, `bootstrap-gcp.conf`, `bootstrap-hashicorp-vault.conf` and `bootstrap-aws.conf` sees to be new files to be able to extract some properties like the sensitive key. So maybe we should add support to them to be able to configure them like the `bootstrap.conf`, no? (Still in a different PR)

Reply:
Yes i think that's a good idea. We can add ways to override these files but default to the NiFi provided settings. This way, we don't need to maintain any templates in nifikop which makes our lives easier. Ideally we can keep nifikop agnostic to what version of NiFi it's deploying as much as possible

Reply:
I removed the `logback.xml` template and made sure that it will only be overridden if it's necessary. I let you check the code change to be sure.